lib/iris/util.py

```diff
@@ -1554,3 +1554,104 @@ def _meshgrid(*xi, **kwargs):
+
+    Args:
+        cube:
+        An instance of :class:`iris.cube.Cube`
+
+    Kwargs:
+        rel_tol:
+        relative tolerance
+
+        abs_tol:
+        absolute tolerance in coordinate units
+
+    Returns:
+        Boolean array representing the truth value for discontiguous cells in
+        the coordinate points array.  This can be used as the input array for
+        :func:`iris.util.mask_discontiguities`.
+
+    Examples::
+
+        # Find any unknown discontiguities in your cube's x and y arrays:
+        discontiguities = iris.util.find_discontiguities(cube)
+
+        # Pass the resultant boolean array to `iris.util.mask_discontiguities`
+        # with a cube slice; this will use the boolean array to mask
+        # any discontiguous data points before plotting:
+        masked_cube_slice = iris.util.mask_discontiguities(cube[0],
+                                                           discontiguities)
+
+        # Plot the masked cube slice:
+        iplt.pcolormesh(masked_cube_slice)
+
+    """
+    lats_and_lons = ['latitude', 'grid_latitude',
+                     'longitude', 'grid_longitude']
+    spatial_coords = [coord for coord in cube.aux_coords if
+                      coord.name() in lats_and_lons]
+    dim_err_msg = 'Discontiguity searches are currently only supported for ' \
+                  '2-dimensional coordinates.'
+    if len(spatial_coords) != 2:
+        raise NotImplementedError(dim_err_msg)
+
+    # Check which dimensions are spanned by each coordinate.
+    for coord in spatial_coords:
+        if coord.ndim != 2:
+            raise NotImplementedError(dim_err_msg)
+        elif isinstance(coord, int):
+            span = set([coord])
+        else:
+            span = set(cube.coord_dims(coord))
+        if not span:
+            msg = 'The coordinate {!r} doesn\'t span a data dimension.'
+            raise ValueError(msg.format(coord.name()))
+
+    # Check that the 2d coordinate arrays are the same shape as each other
+    if len(spatial_coords) == 2:
+        assert spatial_coords[0].points.shape == spatial_coords[1].points.shape
+
+    # Set up unmasked boolean array the same size as the coord points array:
+    bad_points_boolean = np.zeros(spatial_coords[0].points.shape, dtype=bool)
+
+    for coord in spatial_coords:
+        _, (diffs_x, diffs_y) = coord._discontiguity_in_bounds(rtol=rel_tol,
+                                                               atol=abs_tol)
+        gaps_x = diffs_x > abs_tol
+        gaps_y = diffs_y > abs_tol
+
+        bad_points_boolean[:, :-1] = np.logical_or(bad_points_boolean[:, :-1],
+                                                   gaps_x)
+        # apply mask for y-direction discontiguities:
+        bad_points_boolean[:-1, :] = np.logical_or(bad_points_boolean[:-1, :],
+                                                   gaps_y)
+    return bad_points_boolean
+
+
+def mask_discontiguities(cube_slice, discontiguous_points):
```

Comment:
Not clear why this is called 'cube_slice'.  
If you want to make the point that we only support 2d cubes, I don't think this helps much.
Can we call it just "cube" ?

Reply:
Likewise, I wouldn't call the array argument 'discontiguous_points', because it undermines the point of making this a separate function, which is to emphasize that the user is making a definite decision to mask off some points, whether those chosen by the 'find_' routine, or something else.

Reply:
Actually the routine shouldn't be called 'mask_discontiguities' either !!!

Reply:
`def mask_cube(cube, points_to_mask)`